lib/views/login/login_form.dart

```diff
@@ -77,32 +81,73 @@ class LoginFormState extends ConsumerState<LoginForm> {
     );
   }
 
+  @override
+  void initState() {
+    super.initState();
+    _email.addListener(() =>
+        ref.read(loginViewModelProvider.notifier).validateEmail(_email.text));
+    _password.addListener(() => ref
+        .read(loginViewModelProvider.notifier)
+        .validatePassword(_password.text));
+  }
+
+  @override
+  void dispose() {
+    _email.dispose();
+    _password.dispose();
+    super.dispose();
+  }
+
   @override
   Widget build(BuildContext context) {
-    return Column(
-      mainAxisAlignment: MainAxisAlignment.center,
-      crossAxisAlignment: CrossAxisAlignment.stretch,
-      children: [
-        _emailTextField,
-        const SizedBox(height: 20),
-        Stack(
-          alignment: AlignmentDirectional.centerEnd,
-          children: [
-            _passwordTextField,
-            _forgetPasswordButton(context),
-          ],
-        ),
-        const SizedBox(height: 20),
-        _loginButton,
-      ],
+    return Form(
+      key: _loginFormKey,
+      autovalidateMode: _isStartedValidation
+          ? AutovalidateMode.onUserInteraction
+          : AutovalidateMode.disabled,
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          _emailTextField,
+          const SizedBox(height: 20),
+          Stack(
```

Comment:
@nmint8m It would be better if we can move this widget into a separate function, to reduce LOC in `build` function
```suggestion
          Widget _passwordRegion() {
             return  Stack(
                alignment: AlignmentDirectional.centerEnd,
                children: [
                   _passwordTextField,
                   _forgetPasswordButton(context),
            ],
          }
```